### PR TITLE
feat(content-manager): add syntax highlighting in the code block type in the Blocks field original solution

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

Adds the highlight colors in the Code blocks inside Block Editor.

### Why is it needed?

We want to highlight the syntax of every single language.
We used Prismjs to do it and this theme https://prismjs.com/index.html?theme=prism-solarizedlight if we want something else just add a comment.

### How to test it?

 User adds a code block in the Blocks field
 - GIVEN THAT I have selected the code block type,
 - WHEN I enter code into the code block
   - THEN the code is displayed with syntax highlighting according to the language detected or specified

 User selects a specific programming language for syntax highlighting
 - GIVEN THAT I am editing a code block in the Blocks field,
 - WHEN I choose a programming language from a dropdown list (e.g., JavaScript, Python, HTML),
   - THEN the code block applies syntax highlighting specific to the selected language

### Related issue(s)/PR(s)

CS-865

### CODE REVIEW

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/prism.ts
Comment (remidej) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/prism.ts:1]: Is this the fix Alex was recommending? I find it dangerous to be honest. It means we now own that code, we can't get new patch versions, ESLint and Prettier are gonna run against it on all commits, etc. I'd prefer to keep it as a dependency even if that means bypassing the CI on the first PR that introduces it

Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/prism.ts:1]: sorry It was an attempt that I needed to remove

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/Blocks/utils/decorateCode.ts
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/Blocks/utils/decorateCode.ts:1]: ok

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksContent.tsx [hunk 1]
```diff
@@ -358,7 +367,7 @@ const BlocksContent = ({ placeholder, ariaLabelId }: BlocksContentProps) => {
 
   // Create renderLeaf function based on the modifiers store
   const renderLeaf = React.useCallback(
-    (props: RenderLeafProps) => baseRenderLeaf(props, modifiers),
+    (props: RenderLeafProps) => baseRenderLeaf(props as ExtendedRenderLeafProps, modifiers),
```
[the remark this answers is not in the document]
Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksContent.tsx:370]: yes you are right, I can define className prop as optional

File: packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
+  value: string;
+  label: string;
+  decorate?: string;
+  importPath?: string;
+}[] = [
   {
     value: 'asm',
     label: 'Assembly',
+    decorate: 'asmatmel',
+    importPath: 'prism-asmatmel',
```
Comment (remidej) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts:11]: Can you point me to the place in the code that uses this importPath property? I can't seem to find it

Comment (simotae14) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/utils/constants.ts:11]: yeah, you are right, it is a property that I tried to use to lazy load the imports with no success, I am gonna remove them

**Review by remidej — APPROVED** [review]
tested and it works well 💪

**Comment by remidej** [comment]
Can you just double check that all the tests pass locally for you before merging? Since the CI's not running

**Comment by pandeyprafull** [comment]
Screenshot from 2024-12-12 18-26-06
getting the error message whenever try to create new entry of any collection

**Comment by simotae14** [comment]
@pandeyprafull thanks for the message but you need to open a Github issue in the repo with all the details https://github.com/strapi/strapi/issues.
Can I ask you to try to reinstall the dependencies and rerun your Strapi project, thanks

**Comment by pandeyprafull** [comment]
@simotae14 thanx for the suggestion works fine now

**Pull request merged**